src/main/java/org/spongepowered/api/service/stat/StatsConfigManager.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.service.stat;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.plugin.PluginContainer;
+
+/**
+ * Provides information about whether a server has granted permission for
+ * server data to be transmitted on a per plugin basis.
+ */
+public interface StatsConfigManager {
```

Comment:
This should simply be StatsService

Reply:
I would like to reserve the StatsService name the actual statiszic collector service nane

Reply:
I can change it, but I named it a `Manager` because it's not intended to be replacable. It's not registered in the service manager either (though it could be). It also doesn't handle stats itself, just whether the config gives it permission, which is why I went for `StatsConfigManager`. Could go for `StatsPermission` or something if people prefer.

Reply:
If we call it like Permission then why not remove it entirely and use the `PermissionService` with a `SubjectType#PLUGIN` directly?
The plugin id could be the subject id and its even extensible for whatever plugin permissions we are going to add in the future (like `download.file` or whatever).

Even if we dont merge it with the permission service. The more complex/longer the name becomes the more I am in favor of just calling it `PluginPermission...` and allow it to be at least extensible for future additional plugin permissions. This would also make it easier to decide on a package for this class as `plugin` would be a good choice.

Reply:
That's basically going to add a layer onto the steps that server admins would have to follow, and would basically _require_ a permission plugin to opt-in/out. We just need to make it easy for users to opt-in and be done with it.

All this is, and needs to be, is a simple config check. `StatAuthorityManager` would work as a name (as much as I like having the acronym SPAM in the API...), just got to decide the package. Any ideas welcome, I do not have much time to spend on this tonight, so it'll get done over the next day or so.

I'd rather get this straightened out before I have many commits that simply change the name of things.

Reply:
What about putting this in the `util` package? I thought about the `config` package, but that's for managing configs, not exposing Sponge configs.

Bit of a loss on where to put this otherwise.

Reply:
As per yesterday's discussion, we're keeping `StatsConfigManager` I may bend if we want to say `StatCollectionConfigManager` just to make it clearer it's about statistics collection.